src/controllers/StaffController.js

```diff
@@ -553,6 +625,580 @@ class StaffController {
     }
   }
+
+  /**
+   * Retrieves all objects from the specified collection for admin access.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of retrieving all objects.
+   * @throws {Error} If there is an error during the retrieval process.
+   */
+  static async adminGetAll(req, res) {
+    let adminFxn = await authClient.staffPreCheck(req);
+    if (adminFxn.error) {
+      return res.status(401).json({
+        error: adminFxn.error,
+        msg: 'Staff is not authorized to perform this action',
+      });
+    }
+    const { staff, token } = adminFxn;
+    const data = await authClient.staffAttrCheck(req);
+    if (data.error) {
+      return res.status(400).json({ error: data.error });
+    }
+    const typeCollection = data.collection;
+    try {
+      adminFxn = await mongoose.model(typeCollection).find({}).exec();
+      if (adminFxn.error) {
+        return res.status(401).json({
+          error: adminFxn.error,
+        });
+      }
+    } catch (err) {
+      return res.status(400).json({ error: err });
+    }
+    if (!adminFxn) {
+      return res.status(400).json({ error: 'Internal Server Error', msg: 'Failed to get all objects' });
+    }
+    return res.status(200).json({ adminFxn, token });
+  }
+
+  /**
+   * Creates a new object in the specified collection for admin access.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of creating a new object.
+   * @throws {Error} If there is an error during the creation process.
+   */
+  static async adminCreateNew(req, res) {
+    // extract the content of the request
+    const data = await authClient.staffAttrCheck(req);
+    if (data.error) {
+      return res.status(400).json({ error: data.error });
+    }
+    const adminFxn = await authClient.staffPreCheck(req);
+    if (adminFxn.error) {
+      return res.status(401).json({
+        error: adminFxn.error,
+        msg: 'Staff is not authorized to perform this action',
+      });
+    }
+    const { staff, token } = adminFxn;
+    const typeCollection = data.collection;
+    if (!typeCollection) {
+      return res.status(400).json({
+        error: 'Missing collection type for the request',
+        msg: 'Ensure the key is defined as \'collection\'',
+      });
+    }
+    let newObjData;
+    try {
+      newObjData = await staff.createNew(typeCollection, data);
+      if (newObjData.error) {
+        return res.status(400).json({
+          error: newObjData.error,
+        });
+      }
+    } catch (err) {
+      return res.status(400).json({ error: err });
+    }
+    if (!newObjData) {
+      return res.status(400).json({ error: 'Internal Server Error', msg: 'Failed to create new object' });
+    }
+    const operatorData = await staff.getDashboardData();
```

Comment:
This variable should have the same name you defined in the login and account activation routes for consistency.

_This goes for everywhere else you did same in this file. . ._